packages/good-design/src/apps/ubi/context/ClaimContext.tsx

```diff
@@ -0,0 +1,226 @@
+  withNewsFeed: boolean;
+}
+
+export const ClaimProvider: FC<PropsWithChildren<ClaimProviderProps>> = ({
+  children,
+  explorerEndPoints = {
+    MAINNET: "https://api.etherscan.io/api?",
+    CELO: "https://api.celoscan.io/api?",
+    FUSE: "https://explorer.fuse.io/api?"
+  },
+  provider,
+  supportedChains,
+  withSignModals,
+  withNewsFeed = true,
+  onUpgrade,
+  onConnect,
+  onNews,
+  // onSendTx,
+  onSuccess,
+  onSwitchChain
+}) => {
+  const { account, chainId, library, switchNetwork } = useEthers();
+  const [refreshRate, setRefreshRate] = useState<QueryParams["refresh"]>(4);
+  const [preClaimPools, setClaimPools] = useState<any[] | undefined>(undefined);
+  const [postClaimPools, setPostClaimPools] = useState<any[] | undefined>(undefined);
+  const activeChain = SupportedChains[chainId as number] ?? "CELO";
+  const endpoints = explorerEndPoints[activeChain as keyof typeof SupportedChains];
+  const [error, setError] = useState<string | undefined>(undefined);
+  const claimedAlt = useClaimedAlt(chainId);
+
+  const [txDetails, setTxDetails] = useState({ transaction: undefined, isOpen: false });
+
+  const formattedTransactionList = useFormatClaimTransactions(
+    postClaimPools && postClaimPools?.length > 0 ? postClaimPools : preClaimPools ? preClaimPools : [],
+    chainId,
+    account ?? ""
+  );
+
+  const claimDetails = useClaim(refreshRate);
+  const { poolsDetails, loading, fetchPools } = useGetMemberUBIPools();
+
+  const { errorMessage } = { errorMessage: undefined };
+
+  const { poolContracts, startClaiming: onClaim, claimFlowStatus } = useMultiClaim(preClaimPools);
+
+  const switchChain = useCallback(() => {
+    // 4902: Network is not added, and should be done manually
+    // explanation to user is shown through network modal
+    switchNetwork(SupportedChains[claimedAlt.altChain as keyof typeof SupportedChains]).catch((e: any) => {
+      if (e.code === 4902) {
+        // toggleNetworkModal()
+        //todo: discuss how to handle this
+      }
+    });
+  }, [switchNetwork, claimedAlt]);
+
+  const onTxDetailsPress = useCallback(
+    (transaction: any) => {
+      setTxDetails({ transaction, isOpen: true });
+    },
+    [txDetails]
+  );
+
+  const onClaimFailed = useCallback(async () => {
+    setError(errorMessage ?? /*i18n*/ "An unknown error occurred while claiming");
+    setClaimPools(undefined);
+  }, [errorMessage]);
+
+  const onClaimSuccess = useCallback(async () => {
+    // should handle what happens after all claims are done (eg. showing a next-task modal)
+    // if nothing is done, it will just silently finish
+
+    setClaimPools(undefined);
+    setRefreshRate("everyBlock");
+    void fetchPools();
+    await onSuccess?.();
+  }, [onSuccess]);
+
+  useEffect(() => {
+    setClaimPools(undefined);
+    setPostClaimPools(undefined);
+    void fetchPools();
+  }, [/*used*/ chainId, /*used*/ account]);
+
+  //Handling of claimable pools
+  useEffect(() => {
```

Comment:
If you claim from pools and stay on the page.
The next day, will the pool states be refreshed even though you never left the page?

Reply:
fixed applied here: https://github.com/GoodDollar/GoodWeb3-Mono/pull/162/commits/fac8eb603a4259d6689b6d6d6b62d404f95812ac